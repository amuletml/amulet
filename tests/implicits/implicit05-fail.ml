(* Tests the ambiguous type error message *)

type foo 'a = Foo of int

let implicit foo_int : foo int = Foo 1
let implicit foo_string : foo string = Foo 1

let foo : forall 'a. foo 'a => foo 'a = fun x -> x

let main = foo
