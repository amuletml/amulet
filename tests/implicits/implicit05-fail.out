implicit05-fail.ml[10:12 ..10:14]: error
  Could not find implicit value of type foo 'a

  • Ambiguous type variable 'a prevents finding a value
  • Suggestion: use a type annotation to specify it
                perhaps to the type foo int

  These relevant implicit values are in scope: 
  • foo_int : foo int
  • foo_string : foo string

  Arising in the expression
   │ 
10 │ let main = foo
   │ 
